shuup_tests/core/test_rounding.py

```diff
@@ -100,8 +106,52 @@ def test_order_source_rounding(prices):
         # make sure the line taxless price is rounded
         assert order_source.taxless_price == source.shop.create_price(bankers_round(price, 2))
 
+        # Check that total prices calculated from priceful parts still matches
+        assert _get_taxless_price(order_source) == order_source.taxless_price
+        assert _get_taxful_price(order_source) == order_source.taxful_price
+
         # make sure the line price is rounded
         assert order_source.price == source.shop.create_price(price)
 
     # make sure order total is rounded
     assert source.taxless_total_price == source.shop.create_price(bankers_round(expected, 2))
+
+
+@pytest.mark.parametrize("prices", PRICE_SPEC)
+@pytest.mark.django_db
+def test_rounding_with_taxes(prices):
+    shop = get_default_shop()
+    supplier = get_default_supplier()
+    expected = 0
+    for p in prices:
+        expected += bankers_round(p, 2)
+
+    order = create_empty_order(shop=shop)
+    order.save()
+    product = create_product("test_sku",  shop=shop, supplier=supplier)
+    add_product_to_order(
+        order, supplier, product, quantity=Decimal("2.22"),
+        taxless_base_unit_price=Decimal("10"), tax_rate=Decimal("0.22222"))
+    for x, price in enumerate(prices):
+        ol = OrderLine(
+            order=order,
+            type=OrderLineType.OTHER,
+            quantity=1,
+            text="Thingy",
+            ordering=x,
+            base_unit_price=order.shop.create_price(price)
+        )
+        ol.save()
+    order.cache_prices()
+    for x, order_line in enumerate(order.lines.all().order_by("ordering")):
+        # Check that total prices calculated from priceful parts still matches
+        assert _get_taxless_price(order_line) == order_line.taxless_price
+        assert _get_taxful_price(order_line) == order_line.taxful_price
+
+
```

Comment:
i think it would be a good idea to also add tests to ensure all the invariants in `Priceful` are still met (block starting at line 44 in `_priceful.py`)